docs/running-on-yarn.md

```diff
@@ -134,6 +134,13 @@ Most of the configs are the same for Spark on YARN as for other deployment modes
   </td>
 </tr>
 <tr>
+  <td><code>spark.yarn.am.port</code></td>
+  <td>(random)</td>
+  <td>
+    Port for the YARN Application Master to listen on.
```

Comment:
Unless I'm missing something this is only used in yarn client mode.  This the communication channel (rpc) between AM on cluster and driver on the client side.  It looks like the rest of spark uses things just like spark.executor.port, spark.driver.port, etc.. So this kind of follows that convention.  Personally I would like it more descriptive but since that is kind of convention I'm ok with it.